src/libs/Navigation/Navigation.js

```diff
@@ -74,9 +81,18 @@ function closeDrawer() {
  */
 function getDefaultDrawerState(isSmallScreenWidth) {
```

Comment:
I'm not really getting this method or why we are using it...

Found this...

https://github.com/Expensify/App/blob/f2218b6ff6e3f2453dd2f6fa137352899a26c2b9/src/libs/Navigation/AppNavigator/BaseDrawerNavigator.js#L45-L48

Which led me here:

https://github.com/Expensify/App/pull/8067

The comment there makes no sense at all to me and vaguely references the "navigation internal state". @parasharrajat @thienlnam what exactly is this code doing?

Reply:
I do not know the real reason for how this breaks it internally so I used the term `navigation internal state` but I found the following: 

This component uses a key prop to remount the navigator when the browser resizes. This hack was done to prevent the layout from breaking on the drawer. Now due to that, it is somehow breaking the internal navigation state on change of `defaultStatus`. So instead of passing the prop directly to the navigator, I calculated it once per mount and passed. It seems to work at that time. I moved forward with this change as we can't remove the `key hack`. It breaks the UI for Drawer. 

So in short, we can remove the key prop hack, we can get rid of this. 

So as we are trying to move away from custom Logic. I think an audit should be done of our navigation implementation and bad patterns or hacks should be removed. I can take a look at that but not sure if I will be able to suggest a better design.

Reply:
>I do not know the real reason

Ok haha well, if you don't know the reason as the author who will? 😅 

>This component uses a key prop to remount the navigator when the browser resizes. This hack was done to prevent the layout from breaking on the drawer. 

Ok and we are not sure why the layout breaking on the drawer in the first place? Or did we discover the root cause?

>in short, we can remove the key prop hack, we can get rid of this.

Is there any open issue to figure out what the root cause is? Maybe would be good to create one issue that:

- Documents the workaround we took and why with as much information as possible
- Asks to investigate the root cause
- Try to fix it in `react-navigation`

Reply:
Maybe we can create a new issue to get to the root cause of #5591 ? Temporary solution you have there seems fine for now, but could have been documented better.